packages/compartment-mapper/src/node-modules.js

```diff
@@ -117,15 +181,25 @@ const inferParsers = (descriptor, location) => {
   return commonParsers;
 };
 
-// graphPackage and gatherDependency are mutually recursive functions that
-// gather the metadata for a package and its transitive dependencies.
-// The keys of the graph are the locations of the package descriptors.
-// The metadata include a label (which is informative and not necessarily
-// unique), the location of each shallow dependency, and names of the modules
-// that the package exports.
-
+/**
+ * graphPackage and gatherDependency are mutually recursive functions that
+ * gather the metadata for a package and its transitive dependencies.
+ * The keys of the graph are the locations of the package descriptors.
+ * The metadata include a label (which is informative and not necessarily
+ * unique), the location of each shallow dependency, and names of the modules
+ * that the package exports.
+ *
+ * @param {string} name
+ * @param {ReadDescriptorFn} readDescriptor
+ * @param {Graph} graph
+ * @param {Object} packageDetails
+ * @param {string} packageDetails.packageLocation
+ * @param {Object} packageDetails.packageDescriptor
+ * @param {Set<string>} tags
+ * @returns {Promise<Object>}
```

Comment:
Please make more specific, or drop the `@returns` line entirely.  

```suggestion
 * @returns {Promise<Array<Object> | undefined>}
```